lib/common/compiler.h

```diff
@@ -91,18 +91,29 @@
 /* prefetch
  * can be disabled, by declaring NO_PREFETCH macro */
 #if defined(NO_PREFETCH)
-#  define PREFETCH(ptr)   /* disabled */
+#  define PREFETCH(ptr)     (void)(ptr)  /* disabled */
 #else
 #  if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_I86))  /* _mm_prefetch() is not defined outside of x86/x64 */
 #    include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
-#    define PREFETCH(ptr)   _mm_prefetch((const char*)ptr, _MM_HINT_T0)
+#    define PREFETCH(ptr)   _mm_prefetch((const char*)(ptr), _MM_HINT_T1)
 #  elif defined(__GNUC__) && ( (__GNUC__ >= 4) || ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) ) )
-#    define PREFETCH(ptr)   __builtin_prefetch(ptr, 0, 0)
+#    define PREFETCH(ptr)   __builtin_prefetch((ptr), 0 /* rw==read */, 2 /* locality */)
```

Comment:
Sure.
Note that this part of the code is not really "set in stone". It's primarily designed this way because it feels "reasonable" and I could not measure any difference between the different priority levels in my tests. But in theory, one could or should differentiate the type of "prefetch" requested, depending on their potential usage pattern.
This will probably be revisited in the future, when _prefetch_ will be a more central part of the decoding engine.